js/apps/system/_admin/aardvark/APP/react/src/views/collections/indices/addIndex/invertedIndex/useCreateInvertedIndex.ts

```diff
@@ -0,0 +1,320 @@
+import { useCreateIndex } from "../useCreateIndex";
+
+/**
+ * Index definition could be split on several blocks.
+ * 
+ * 1. Commit/Consolidation properties - same as for ArangoSearch Views:  could be found here https://www.arangodb.com/docs/stable/arangosearch-views.html. Starting from “cleanupIntervalStep“ and below. Mutable!  For this I suggest to have something similar to what we already have for ArangoSearch Views.
+
+ * 2. “root“ fields properties: 
+```
+  analyzer: <analyzerName:string>,
+  features: <analyzerFeatures: array, possible values [ "frequency", "position", "offset", "norm"], optional, default []>,
+  includeAllFields: <bool, optional, default: false>,
+  trackListPositions: <bool, optional, default: false>,
+  searchField: <bool, optional, deafult: false>
+```
+ * 3. Fields definition.  In CE this is just an array of objects. But for EE we have “nested” fields with arbitrary nesting depth.  That is an array of fields objects.
+    ```
+        {
+          name: <name:string>,
+          analyzer: <analyzerName:string, optional>
+          searchField: <bool, optional>
+          features: <analyzerFeatures: array, possible values [ "frequency", "position", "offset", "norm"], optional, default []>   
+          includeAllFields: <bool, optional>
+          trackListPositions: <bool, optional>          
+          nested: [
+            // see below, EE only, optional
+            ],
+          },
+        }
+```
+    And nested fields definitons looks like - same nested structure as we have had for ArangoSearch views fields in general.
+```
+  nested: 
+        [ { "name":"subfoo",
+              analyzer: <analyzerName:string, optional>,
+              searchField: <bool, optional>,
+              features: <analyzerFeatures: array, possible values [ "frequency", "position", "offset", "norm"], optional, default []>
+              nested: [
+                //sub-nested fields (optional)
+              ]
+             },
+            .....
+          ]
+```
+ * 4.Primary sort. Very like ArangoSearch views primary sort but defined a bit different - compression is moved inside primarySort object. Potentially there would be more fields.  
+```
+  primarySort: { fields: [
+    // fields array as for arangosearch views
+  ], compression: lz4}
+```
+*/
+type BytesAccumConsolidationPolicy = {
+  type: string;
+  threshold?: number;
+};
+
+type TierConsolidationPolicy = {
+  type: string;
+  segmentsMin?: number;
+  segmentsMax?: number;
+  segmentsBytesMax?: number;
+  segmentsBytesFloor?: number;
+  minScore?: number;
+};
+
+type ConsolidationPolicy =
+  | TierConsolidationPolicy
+  | BytesAccumConsolidationPolicy;
+
+type AnalyzerFeatures = "frequency" | "position" | "offset" | "norm";
+
+type PrimarySortFieldType = {
+  field: string;
+  direction: "asc" | "desc";
+};
+
+export type InvertedIndexFieldType = {
+  name: string;
+  analyzer?: string;
+  features?: AnalyzerFeatures[];
+  searchField?: boolean;
```

Comment:
```suggestion
  shouldSearchField?: boolean;
```

This kind of feels like the string of the search field currently.

Reply:
I agree, `searchField` sounds like it should be a `string`, but since this is the key on backend, we could change on just FE, but will need to convert back. I think we should leave it like this unless backend also wants to fix it. What do you think?

Also, cc: @Dronplane for feedback on the field name